src/ivq.tsx

```diff
@@ -1,30 +1,69 @@
 import {h} from 'preact';
-import { IvqConfig } from "./types/IvqConfig";
+import {QuizLoader} from './providers/quiz-loader';
+import {IvqConfig} from './types/IvqConfig';
 
-// @ts-ignore
 export class Ivq extends KalturaPlayer.core.BasePlugin {
   private _player: KalturaPlayerTypes.Player;
+  private _resolveReadyPromise = () => {};
+  private _readyPromise: Promise<void>;
 
-  static defaultConfig: IvqConfig = {
-
-  };
+  static defaultConfig: IvqConfig = {};
 
   constructor(name: string, player: any, config: IvqConfig) {
     super(name, player, config);
     this._player = player;
+    this._readyPromise = this._makeReadyPromise();
   }
 
   get ready() {
-    return Promise.resolve();
+    return this._readyPromise;
   }
 
-  loadMedia(): void {}
+  loadMedia(): void {
+    const kalturaCuePointService: any = this._player.getService('kalturaCuepoints');
+    this._getQuestionsAndAnswers(kalturaCuePointService);
+    this._getQuiz();
+  }
 
   static isValid(): boolean {
     return true;
   }
 
-  reset(): void {}
+  reset(): void {
+    this._readyPromise = this._makeReadyPromise();
+  }
 
   destroy(): void {}
+
+  private _makeReadyPromise = () => {
+    return new Promise<void>(res => {
+      this._resolveReadyPromise = res;
+    });
+  };
+
+  private _getQuestionsAndAnswers(kalturaCuePointService: any) {
```

Comment:
I think we need to make some change in name convention so it is more clear - maybe call the meta data answers optionalAnswers to separate them from the user answered ones.